c/zss.c

```diff
@@ -389,6 +392,40 @@ static JsonObject *readServerSettings(ShortLivedHeap *slh, const char *filename)
   return mvdSettingsJsonObject;
 }
 
+static hashtable *getServerUserTimeoutsHt(ShortLivedHeap *slh, const char *filename, const char *key) {
+
+  char jsonErrorBuffer[512] = { 0 };
+  int jsonErrorBufferSize = sizeof(jsonErrorBuffer);
+  Json *serverTimeouts = NULL; 
+  JsonObject *serverTimeoutsJsonObject = NULL;
+  hashtable *ht = NULL;
+  zowelog(NULL, LOG_COMP_ID_MVD_SERVER, ZOWE_LOG_DEBUG, "reading '%s' timeout settings from %s\n", key, filename);
+  serverTimeouts = jsonParseFile(slh, filename, jsonErrorBuffer, jsonErrorBufferSize);
+  if (serverTimeouts) {
+    if (jsonIsObject(serverTimeouts)) {
+      serverTimeoutsJsonObject = jsonAsObject(serverTimeouts);
+    } else {
+      zowelog(NULL, LOG_COMP_ID_MVD_SERVER, ZOWE_LOG_SEVERE, ZSS_LOG_FILE_EXPECTED_TOP_MSG, filename);
+    }
+    JsonObject *users = jsonObjectGetObject(serverTimeoutsJsonObject, key);
+    if (users) {
+      JsonProperty *property = jsonObjectGetFirstProperty(users);
+      while (property != NULL) {
+        char *userKey = jsonPropertyGetKey(property);
+        int timeoutValue = jsonObjectGetNumber(users, userKey);
+
+        ht = htCreate(500, NULL, NULL, NULL, NULL); // Going to use natural hash function
```

Comment:
`backboneSize` should be a prime number, like 127.
As you use strings as keys in the hashtable you must use `stringHash` and `stringCompare`.
For example:
```C
htCreate(127, stringHash, stringCompare, NULL, NULL);
```